### DISCUSSION TITLE
Conditional Rules

### DISCUSSION THREAD

**Discussion by larsontim12** [microsoft/RulesEngine#84] [Q&A] [answered]
How would I go about creating conditional rules.  I have a different set of rules that need to run for certain states. Like show below.  Is this possible if so what would it look like?

if(input1.State == 'CA')
{
   Rule1
   Rule2
}

if(input1.State == 'WA')
{
   Rule1
   Rule2
}

Thanks,
Tim

**Comment by abbasc52** [comment, marked as answer]
Hi @larsontim12,
There is no direct feature currently to conditionally run rules.

Currently workflow define a set of Rules and you can execute a workflow by its name.
So if you could separate the Rules into multiple workflows, you could achieve the above ask

```jsonc
[{
 "WorkflowName": "CA",
  Rules:[
    Rule1,Rule2
  ]
},
{
"WorkflowName": "WA",
  Rules:[
    Rule2,Rule3
  ]
}
]
```

Then you could run Rulesengine with the appropriate workflow:
```csharp
       var result = await re.ExecuteAllRulesAsync(input1.State, input1);
```

Let me know your thoughts on this
